### PR TITLE
Code clean-up

### PR DESCRIPTION
**Pull request by krmahadevan**
* Use wild card as types instead of raw param type for Class
* Use String instead of StringBuilder when trying to build a 
String wherein there is no iterations
* Favour streams over iterators
* don’t use “newInstance()”

### CODE REVIEW

**Review by baubakg — CHANGES_REQUESTED** [review]

File: src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java [hunk 1]
```diff
@@ -555,12 +559,9 @@ protected static Properties importPhaseData() {
...
 
-            StringBuilder lt_sb = new StringBuilder(STD_PHASED_GROUP_PREFIX);
-            lt_sb.append(lt_nrBeforePhase);
-            lt_sb.append("_");
-            lt_sb.append(lt_nrAfterPhase);
-
-            l_objectArrayPhased[rows][0] = lt_sb.toString();
+            l_objectArrayPhased[rows][0] = STD_PHASED_GROUP_PREFIX + lt_nrBeforePhase
+                + "_"
+                + lt_nrAfterPhase;
```
Comment (baubakg) [inline src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java:564]: @krmahadevan Coulld you explain the reason for this change?

Comment (krmahadevan) [reply src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java:564]: @baubakg - Best use case for using StringBuilder is when you are building a string incrementally in a for loop. That's not the case here. So there's no reason why we need to be using a StringBuilder. You can see these sort of suggestions from IntelliJ as well.

File: src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java [hunk 2]
```diff
@@ -1187,24 +1182,23 @@ public static Boolean isTestsSelectedByProducerMode() {
...
+
+        l_dataProviderMethod.setAccessible(true);
 
         try {
-            return (Object[][]) l_dataProviderMethod.invoke(l_dataProviderClass.newInstance(), new Object[0]);
-        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException
-                | InstantiationException e) {
+            Object obj = l_dataProviderClass.getDeclaredConstructor().newInstance();
+            return (Object[][]) l_dataProviderMethod.invoke(obj, new Object[0]);
+        } catch (Exception e) {
```
[the remark this answers is not in the document]
Comment (baubakg) [reply src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java:1201]: Yes that is what I am suggesting. I think in general I avoid catching exception as a rule :) However if you have some data on what you are suggesting, please share.

However in this case it seems that the changes you introduce have left the following line uncovered. (Hence the coverage failure).

**Review by baubakg — COMMENTED** [review]
@krmahadevan It seems that a lot of the changes are replacing stringbuilder with simple string additions. It does not seem necessary.

**Comment by krmahadevan** [comment]
> @krmahadevan It seems that a lot of the changes are replacing stringbuilder with simple string additions. It does not seem necessary.

Why would you feel that they aren't necessary? Just curious. They arent breaking any functionality and from what I have read, they are supposed to be the way of forming strings ( sonarcube seems to go to the next level and suggest that one use String.format() for forming the strings )

File: src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (krmahadevan) [reply src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java:1201]: > Yes that is what I am suggesting. I think in general I avoid catching exception as a rule :) However if you have some data on what you are suggesting, please share.

Theres no data that i have that i can share. But i do know that those are the only exceptions that are going to be thrown and we are just wrapping it by adding it as an exception. 



> However in this case it seems that the changes you introduce have left the following line uncovered. (Hence the coverage failure).

Sure. I personally follow The philosophy that when it comes to sonarcube violations and code coverages which is "address them when its going to add value". 

In this case I am not quite sure on the value add, unless and until we decide to abstract out reflection based calls into a helper of its own and write plain vanilla unit tests for them.

You are the repo owner so I leave it to you as the final decision ☺️

[the remark this answers is not in the document]
Comment (baubakg) [reply src/main/java/com/adobe/campaign/tests/integro/phased/PhasedTestManager.java:1201]: Lets put the original exceptions back, as it seems to make the patch coverage to fail.

**Comment by baubakg** [comment]
@krmahadevan I think once we have corrected the patch coverage, we are good to go.

**Comment by krmahadevan** [comment]
@baubakg - I have reverted out the reflection changes but still the codecov check is failing (which doesn't make sense to me). 
I wont be able to dedicate any more time on this PR.
IMO there's no more changes that are needed. 
I leave it to you to accept this PR or reject it.

**Comment by baubakg** [comment]
Faire enough. Needed to be sure. Thanks for all the help. I have and will learn a lot from these changes. 😊

**Pull request merged**